internal/wasm/store_test.go

```diff
@@ -202,29 +199,48 @@ func TestStore_concurrent(t *testing.T) {
 		},
 	}
 
-	// Concurrent instantiation.
-	wg.Add(goroutines)
-	for i := 0; i < goroutines; i++ {
-		go func(i int) {
-			defer wg.Done()
-			_, err := s.Instantiate(context.Background(), importingModule, strconv.Itoa(i), nil)
-			require.NoError(t, err)
-		}(i)
+	// Similar to golang internals, use up to 8 goroutines per hammer test.
+	P := 8
+	// Inside each goroutine, use a large loop in an attempt to trigger a concurrency bug.
+	N := 1000 // Remember, there are many hammer tests, so watch the scale of this as it effects PR validation time.
+	if testing.Short() {
+		P = 4
+		N = 100
 	}
-	wg.Wait()
-
-	// Concurrent release.
-	wg.Add(goroutines)
-	for i := 0; i < goroutines; i++ {
-		go func(i int) {
-			defer wg.Done()
-			require.NoError(t, s.CloseModule(strconv.Itoa(i)))
-			require.NoError(t, err)
-		}(i)
+	defer runtime.GOMAXPROCS(runtime.GOMAXPROCS(P))
```

Comment:
maybe half to ensure that we actually need goroutine switch? 

```suggestion
	defer runtime.GOMAXPROCS(runtime.GOMAXPROCS(P/2))
```

Reply:
cool. I'll comment that also :D